### DISCUSSION TITLE
Division by zero when deleting a device

### DISCUSSION THREAD

**Discussion by knebb** [netbox-community/netbox#17866] [General]
Hi,

II have a typical setup

switch <--> patchpanel <--> wallmounted plug <--> device (like pc)

From the switch I can easily see the cable trace and even see what device is connected to a specific port. Great work, guys!

Now when I just delete the device (without removing the cable first) I am getting an error when I try to trace the cable from the switch as before.

This is what I am getting when trying to trace the cable.
```
<class 'ZeroDivisionError'>

division by zero

Python-Version: 3.11.2
NetBox-Version: 4.1.3
Plugins: 
  netbox_topology_views: 4.0.1
```
So I have to manually trace the cable to the wallmounted plug, remove the cable (which physically does not exist any more) and then I can trace the other cables from the switch port to the plug.

In my expectation it should ignore the cable with the loose end and display the trace as far as possible.

Or is this expected?

/KNEBB

**Comment by candlerb** [comment]
An exception is not expected and should not be visible to end users, so it's a bug. If you can set DEBUG=True in configuration.py, and show the full backtrace from the exception, that would be helpful to the developers.

> I have to manually trace the cable to the wallmounted plug, remove the cable (which physically does not exist any more) ... In my expectation it should ignore the cable with the loose end

I think that's a situation you shouldn't be able to get into, but people *have* observed this in the past - indeed someone wrote a [script](https://github.com/netbox-community/customizations/blob/master/scripts/find_orphaned_cables.py) to find orphaned cables which are terminated only at one end.

Hence there are potentially two bugs here:
1. You've found a way to end up with a cable which is terminated at one end only. Please provide the *exact* reproduction steps and submit as a bug.
2. In the situation generated by (1), an exception is generated from a cable trace. Ditto.

It's possible that the Netbox dev team will only accept one of these as a bug, as I don't know what the official policy is on the validity of cables which are terminated at one end (although I note the GUI doesn't allow a cable to be explicitly disconnected at one end)

**Reply by knebb** [reply]
According to the list of open issues it is a well-known problem and it appears it is somehow accepted to have cables with only a single termination. See [here](https://github.com/netbox-community/netbox/issues/10769).

However, if a cable with a single termination is accepted as valid the cable trace should take care of it.
Created a new [bug report](https://github.com/netbox-community/netbox/issues/17868) for this.

Oh, BTW: Thanks you for linking the script. I was not aware of having this possibility. Tracked (and removed) all my orphaned or only half-terminated cables now.
